src/main/resources/templates/db_scripts/transactions.sql

```diff
@@ -0,0 +1,109 @@
+-- https://www.postgresql.org/docs/10/functions-json.html
+CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.recalculate_totals() RETURNS TRIGGER AS $recalculate_totals$
+  DECLARE
+    fromBudgetRecord        record;
+    fromBudget              jsonb;
+    fromBudgetAllocated     decimal;
+    fromBudgetAvailable     decimal;
+    fromBudgetUnavailable   decimal;
+
+    fromLedgerRecord        record;
+    fromLedger              jsonb;
+    fromLedgerAllocated     decimal;
+    fromLedgerAvailable     decimal;
+    fromLedgerUnavailable   decimal;
+
+    toBudgetRecord          record;
+    toBudget                jsonb;
+    toBudgetAllocated       decimal;
+    toBudgetAvailable       decimal;
+    toBudgetUnavailable     decimal;
+
+    toLedgerRecord          record;
+    toLedger                jsonb;
+    toLedgerAllocated       decimal;
+    toLedgerAvailable       decimal;
+    toLedgerUnavailable     decimal;
+
+    fromFundRecord          record;
+    toFundRecord            record;
+    newBudgetValues         text[];
+    newLedgerValues         text[];
+    amount                  decimal;
+
+    BEGIN
+      amount = (NEW.jsonb->>'amount')::decimal;
+
+      -- Recalculate Allocation
+      IF (NEW.jsonb->>'transactionType' = 'Allocation') THEN
+        --check if fromFundId exists
+        IF (NEW.jsonb->'fromFundId' IS NOT NULL) THEN
+          -- Update Budget identified by the transactions fiscal year (fiscalYearId) and source fund (fromFundId)
+          SELECT INTO fromBudgetRecord * FROM  ${myuniversity}_${mymodule}.budget WHERE (jsonb->>'fiscalYearId' = NEW.jsonb->>'fiscalYearId' AND jsonb->>'fundId' = NEW.jsonb->>'fromFundId');
+          fromBudget = fromBudgetRecord.jsonb::jsonb;
+
+          fromBudgetAllocated = (SELECT COALESCE(fromBudget->>'allocated', '0'))::decimal - amount;
+          fromBudgetAvailable = (SELECT COALESCE(fromBudget->>'available', '0'))::decimal - amount;
+          fromBudgetUnavailable = (SELECT COALESCE(fromBudget->>'unavailable', '0'))::decimal + amount;
+          newBudgetValues = '{allocated,' || fromBudgetAllocated || ', available, ' || fromBudgetAvailable || ', unavailable, ' || fromBudgetUnavailable ||'}';
+
+          UPDATE ${myuniversity}_${mymodule}.budget SET jsonb = jsonb || json_object(newBudgetValues)::jsonb WHERE (jsonb->>'fiscalYearId' = NEW.jsonb->>'fiscalYearId' AND jsonb->>'fundId' = NEW.jsonb->>'fromFundId');
+
+
+          -- Update Ledger identified by the transaction's fiscal year (fiscalYearId) and the source fund (fromFundId)
+          SELECT INTO fromFundRecord * FROM ${myuniversity}_${mymodule}.fund WHERE (jsonb->>'id' = fromBudget->>'fundId');
+          SELECT INTO fromLedgerRecord * FROM  ${myuniversity}_${mymodule}.ledger  WHERE (jsonb->>'id' = fromFundRecord.jsonb::jsonb->>'ledgerId');
+          fromLedger = fromLedgerRecord.jsonb::jsonb;
+
+          fromLedgerAllocated = (SELECT COALESCE(fromLedger->>'allocated', '0'))::decimal - amount;
```

Comment:
According to jira description the ledger update should be:
> If "fromFundId" is present, recalculate the allocated (increases by transaction amount), available (decreases by transaction amount), unavailable (increases by the transaction amount) fields of the ledger

I believe `+` should be here